ucp/_libs/ucx_api.pyx

```diff
@@ -139,39 +141,43 @@ cdef class UCXContext:
             logging.info("  %s: %s" % (k, v))
 
     def close(self):
-        if self._initialized:
-            self._initialized = False
+        if self.initialized:
+            self.initialized = False
             ucp_cleanup(self._handle)
 
     def get_config(self):
         return self._config
 
     @property
     def handle(self):
+        assert self.initialized
         return int(<uintptr_t><void*>self._handle)
 
 
 cdef class UCXWorker:
     """Python representation of `ucp_worker_h`"""
     cdef:
+        object __weakref__
         ucp_worker_h _handle
-        bint _initialized
         UCXContext _context
+    cdef public:
+        bint initialized
```

Comment:
Same question as in `UCXContext`.

Reply:
It is used by [`ProgressTask()`](https://github.com/madsbk/ucx-py/blob/app_ctx_finalizer/ucp/continuous_ucx_progress.py#L9).

Reply:
This is probably more semantic, but it seems that it's used read-only, right? Would it make sense to still prefix it with `_` to identify that shouldn't be modified? Maybe a `isInitialized` method would be even better.

Reply:
Well if it's `readonly`, Cython has a way to [declare that]( https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#readonly ). So maybe we should just mark it as `readonly`?

Reply:
Trying to address with PR ( https://github.com/rapidsai/ucx-py/pull/472 ). Please take a look 🙂